src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,256 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.service.model.User;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static final int AGE_OF_TWO = 2;
+    private static final int AGE_OF_FIVE = 5;
+    private static final int AGE_OF_TEN = 10;
+    private static final int AGE_OF_EIGHTEEN = 18;
+    private static final int AGE_OF_TWENTY = 20;
+    private static final int AGE_OF_FORTY = 40;
+    private static final int NEGATIVE_AGE_OF_MINUS_FIVE = -5;
+    private static final int NEGATIVE_AGE_OF_MINUS_TWENTY = -20;
+    private static final int NEGATIVE_AGE_OF_MINUS_FIFTY = -50;
+    private static final String GOOD_LOGIN_FIRST = "login12345";
+    private static final String GOOD_LOGIN_SECOND = "admin12345";
+    private static final String GOOD_LOGIN_THIRD = "superAdmin12345";
+    private static final String GOOD_LOGIN_FORTH = "superlogin12345";
+    private static final String GOOD_LOGIN_FIFTH = "admin12345fff";
+    private static final String GOOD_LOGIN_SIXTH = "superAdmin12345SDVDFA";
+    private static final String GOOD_LOGIN_SEVENTH = "login12345sdsfvs";
+    private static final String GOOD_LOGIN_EIGHT = "admin123457e7ch";
+    private static final String GOOD_LOGIN_NINTH = "superAdmin12345asadvc";
+    private static final String GOOD_LOGIN_TENTH = "superAdmin12345asadvcDSSDCADS";
+    private static final String GOOD_PASSWORD_FIRST = "password12345";
+    private static final String GOOD_PASSWORD_SECOND = "password3888112";
+    private static final String GOOD_PASSWORD_THIRD = "passwordMkH61345555";
+    private static final String EMPTY_LOGIN_OR_PASSWORD = "";
+    private static final String LOGIN_LENGTH_OF_TWO = "1h";
+    private static final String LOGIN_LENGTH_OF_FOUR = "ABCD";
+    private static final String PASSWORD_LENGTH_OF_THREE = "j2k";
+    private static final String PASSWORD_LENGTH_OF_FIVE = "KHw7R";
+    private static RegistrationServiceImpl registrationService;
+
+    @BeforeAll
+    static void createReference() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @Test
+    void register_userAlreadyExist_notOk() {
+        User user = new User();
+        user.setLogin(GOOD_LOGIN_FIRST);
+        user.setPassword(GOOD_PASSWORD_FIRST);
+        user.setAge(AGE_OF_TWENTY);
+        Storage.people.add(user);
```

Comment:
Use low-level methods like `Storage.people.add()` directly instead of `RegistrationService.register()` method to add users for testing. This is because you cannot guarantee the correctness of the `register` method before it's been tested.